arm9/src/unk_02002F08.c

```diff
@@ -0,0 +1,125 @@
+#include "global.h"
+#include "unk_02002F08.h"
+#include "font.h"
+#include "gf_gfx_loader.h"
+#include "string16.h"
+#include "unk_0201B8B8.h"
+
+THUMB_FUNC int FUN_02002F08(u32 param0, struct String *str, u32 param2)
```

Comment:
does this match with s32?

Reply:
Yes but `StringGetWidth` and `StringGetWidth_SingleLine_HandleClearToControlCode` return an `int` so I used the same return type

Reply:
do they match with s32s?

Reply:
Yeah let's just change everything